src/main/java/com/emc/ecs/sync/storage/azure/AzureBlobStorage.java

```diff
@@ -0,0 +1,341 @@
+                @Override
+                public InputStream get() {
+                    return readDataStream(cloudBlob);
+                }
+            };
+
+            //TODO how to get ACL for BLOB
+
+            object.setLazyStream(lazyStream);
+            return object;
+        }
+    }
+
+    private List<BlobSyncObject> loadSnapshots(final String key) {
+        List<BlobSyncObject> snapshots = new ArrayList<>();
+
+        for (ListBlobItem blobItem: container.listBlobs(key, true, EnumSet.of(BlobListingDetails.SNAPSHOTS), null, null)) {
+            CloudBlob blob = (CloudBlob) blobItem;
+            ObjectMetadata metadata = syncMetaFromBlobProperties(blob.getProperties());
+            BlobSyncObject object = new BlobSyncObject(this, getRelativePath(key, metadata.isDirectory()), metadata);
+            String snapshotId = "latest.version";
+            if (blob.isSnapshot()) {
+                snapshotId = blob.getSnapshotID();
+            }
+
+            object.setSnapshotId(snapshotId);
+            LazyValue<InputStream> lazyStream = new LazyValue<InputStream>() {
+                @Override
+                public InputStream get() {
+                    return readDataStream(blob);
+                }
+            };
+            object.setLazyStream(lazyStream);
+            object.getDataStream();
+            snapshots.add(object);
+        }
+        log.debug("total blob {} objects(including snapshots) of blob : {}", key ,snapshots.size());
+        return snapshots;
+    }
+
+    @Override
+    public void updateObject(final String identifier, SyncObject object) {
+        //since this plugin would not set as target, no need to override it now
+    }
+
+    private CloudBlob getCloudBlobReference(final String key) {
+        return time(() -> {
+            try {
+                return container.getBlobReferenceFromServer(key);
+            } catch (URISyntaxException | StorageException e) {
+                throw new RuntimeException("Can not get blob reference from server for blob: " + key);
+            }
+        }, OPERATION_GET_BLOB_REFERENCE);
+    }
+
+    private ObjectMetadata syncMetaFromBlobProperties(BlobProperties properties) {
+        ObjectMetadata metadata = new ObjectMetadata();
+
+        metadata.setDirectory(false);
+        metadata.setCacheControl(properties.getCacheControl());
+        metadata.setContentLength(Long.MAX_VALUE);
+        metadata.setContentDisposition(properties.getContentDisposition());
+        metadata.setContentEncoding(properties.getContentEncoding());
+        if (properties.getContentMD5() != null) { metadata.setChecksum(new Checksum("MD5", properties.getContentMD5())); }
+        metadata.setContentType(properties.getContentType());
+        metadata.setModificationTime(properties.getLastModified());
+        metadata.setBlobObjectLength(properties.getLength());
+
+        return metadata;
+    }
+
+    private InputStream readDataStream(final CloudBlob blob) {
```

Comment:
Might want to rename this to `getDataStream()` to avoid confusion.  You aren't actually reading anything (that will be done by the target plugin).